framework/test/jest/specs/unit/controller/application.spec.js

```diff
@@ -0,0 +1,145 @@
+const Application = require('../../../../../src/controller/application');
+const validator = require('../../../../../src/controller/helpers/validator');
+const applicationSchema = require('../../../../../src/controller/schema/application');
+const constantsSchema = require('../../../../../src/controller/schema/constants');
+const version = require('../../../../../src/version');
+
+jest.mock('../../../../../src/controller/helpers/validator');
+
+// Arrange
+const params = {
+	label: '#LABEL',
+	genesisBlock: {},
+	constants: {},
+	config: { components: { logger: null } },
+};
+
+describe('Application', () => {
+	describe('#constructor', () => {
+		it('should accept function as label argument', () => {
+			// Arrange
+			const labelFn = () => '#LABEL';
+
+			// Act
+			const app = new Application(
+				labelFn,
+				params.genesisBlock,
+				params.constants,
+				params.config
+			);
+
+			expect(app.label).toBe(labelFn());
+		});
+
+		it('should set filename for logger if logger component was not provided', () => {
+			// Arrange
+			const config = { components: {} };
+
+			// Act
+			// eslint-disable-next-line no-unused-vars
+			const app = new Application(
```

Comment:
Do we need the `const app` here?

Reply:
Same for other declarations of app that are unused.

Reply:
@shuse2 @michielmulders I think https://eslint.org/docs/rules/no-new rule was complaining when I did that in lisk-modular repository. However, it seems like no-new rule is disabled in this repository. 

I can change it but personally I feel bad when we create an instance from a class without using it. Maybe structure of `application.js` should change and initialisations with side effects should be moved to another method and until then we keep `application.spec.js` as it is, what do you think? see: 
https://stackoverflow.com/questions/2594415/is-it-bad-practice-to-not-assign-a-new-object-to-a-variable

But I updated action.spec.js and event.spec.js since I was just checking if constructor throws exceptions so we don't need the variables there.